src/SetTypeTrait.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Phlib\Mail;
+
+/**
+ * Trait to enable the content-type to be overridden
+ *
+ * @package Phlib\Mail
+ */
+trait SetTypeTrait
```

Comment:
Looking at it a bit further, could you confirm my thinking:

It feels like there shouldn't be a setType method. The only type you want to specify it is on the Mime class. Creating a `new Mime('something/unusual')` class with other subtypes having types defined in the class? Then no `setType` method needed.